gninasrc/gninavis/cnn_visualization.cpp

```diff
@@ -363,76 +445,126 @@ void cnn_visualization::write_scores(const std::vector<float> scores,
+        }
+        else
+        {
+            curr_out_file.open(extended_file_name);
+        }
 
-    out_file << "VIS METHOD: " << method << '\n';
-    out_file << "CNN MODEL: " << cnnopts.cnn_model << '\n';
-    out_file << "CNN WEIGHTS: " << cnnopts.cnn_weights << '\n';
+        curr_out_file << "VIS METHOD: " << method << '\n';
+        if(method == "masking")
+        {
+            curr_out_file << "MASKING TARGET: " << visopts.target << '\n';
+        }
+        if(method == "gradient" || method == "lrp")
+        {
+            curr_out_file << "LAYER IGNORED: " << visopts.layer_to_ignore << '\n';
+        }
 
-    std::stringstream mol_stream(mol_string);
-    std::string line;
-    std::string index_string;
-    int atom_index;
-    std::stringstream score_stream;
-    std::string score_string;
+        if(visopts.target == "pose")
+        {
+            curr_out_file << "POSE SCORE: " << original_score << '\n';
+        }
+        if(visopts.target == "affinity")
+        {
+            curr_out_file << "AFFINITY SCORE: " << original_score << '\n';
+        }
+        else
+        {
+            curr_out_file << "CNN SCORE: " << original_score << '\n';
+        }
 
-    //TODO: don't assume default map!
-    vector<int> typemap; //map atom types to position in grid vectors
-    if(isRec)
-      GridMaker::createDefaultRecMap(typemap);
-    else
-      GridMaker::createDefaultLigMap(typemap);
-
-    float score_sum = 0;
-    int Hadjust = 0; //number of hydrogen atoms seen
-    //dkoes - this is ridiculously fragile, we are assuming atoms don't change
-    //order from the input string, and have to manually adjust for hydrogens
-    //ideally, the model would output this, but it currently doesn't preserve enough
-    //of the receptor :-(
-    //I am also less than fond of the index by 1 approach
-    while (std::getline(mol_stream, line)) {
-        if ((line.find("ATOM") < std::string::npos)
-                || (line.find("HETATM") < std::string::npos)) {
-            score_stream.str(""); //clear stream for next score
-            index_string = line.substr(6, 5);
-            std::string elem = line.substr(77,2);
-            atom_index = std::stoi(index_string);
+        curr_out_file << "MODEL: " << cnnopts.cnn_model << '\n';
+        curr_out_file << "WEIGHTS: " << cnnopts.cnn_weights << '\n';
+
+        std::stringstream mol_stream(mol_string);
+        std::string line;
+        std::string index_string;
+        int atom_index;
+        std::stringstream score_stream;
+        std::string score_string;
+
+        float score_sum = 0;
+        int found_count = 0;
+        std::unordered_map<std::string, bool> found_scores;
+        for(auto i: scores)
+        {
+            found_scores[i.first] = false;
+        }
 
-            parsed_atom pa = parse_pdbqt_atom_string(line);
+        while (std::getline(mol_stream, line)) {
+            if ((line.find("ATOM") < std::string::npos)
```

Comment:
ATOM and HETATM will be at start of string
boost::algorithm::starts_with would be more efficient